src/CanvasRenderingContext2d.cc

```diff
@@ -1269,41 +1283,74 @@ NAN_METHOD(Context2d::DrawImage) {
   // Start draw
   cairo_save(ctx);
 
-  // Scale src
-  float fx = (float) dw / sw;
-  float fy = (float) dh / sh;
+  cairo_matrix_t matrix;
+  double transforms[6];
+  cairo_get_matrix(context->context(), &matrix);
+  decompose_matrix(matrix, transforms);
+  // extract the scale value from the current transform so that we know how many pixels we
+  // need for our extra canvas in the drawImage operation.
+  double current_scale_y = transforms[1];
+  double current_scale_x = transforms[2];
```

Comment:
Are these flipped? Your code comment above says `1: scaleX, 2: scaleY`.

Likewise I think one of the comments on line 1296 or 1297 is supposed to say `Y`.

---

I like where this is headed (a systematic approach as opposed to special handling of individual cases) 👍 . Will give a closer read-through once the above are clarified.

Reply:
yeah i flipped those, trying just with standard scaling i did not notice.